engine/api/src/main/scala/pl/touk/nussknacker/engine/api/component/ComponentProvider.scala

```diff
@@ -0,0 +1,116 @@
+package pl.touk.nussknacker.engine.api.component
+
+import com.typesafe.config.Config
+import pl.touk.nussknacker.engine.api.process.{ProcessObjectDependencies, SingleNodeConfig}
+
+
+/*
+  Service, SourceFactory, SinkFactory, CustomStreamTransformer
+ */
+trait Component
+
+
+/*
+  componentProviders {
+    sqlHsql {
+      provider: sql
+      jdbcUrl: jdbc:hsql...//
+      admin/pass
+    }
+    sqlOracle {
+      provider: sql
+      jdbcUrl: jdbc:oracle...//
+      admin/pass
+    }
+    prinzH20 {
+      provider: prinzH20
+      h2oLocation:
+    }
+    prinzMlFlow {
+      //this is not needed, as configuration is named just as provider
+      //provider: prinzMLFlow
+      mlFlowUrl:
+    }
+    openapiService1 {
+      provider: openapi
+      url: http://host1
+    }
+    openapiService2 {
+      provider: openapi
+      url: http://host2
+      componentsConfig {
+        //we want to ignore particular service here
+        serviceA {
+          disabled: true
+        }
+      }
+    }
+    //not needed in our domain
+    aggregation {
+      disabled: false
+    }
+    configurableFilter {
+      componentsConfig: {
+        filterWithA {
+          component: customFilter
+          customProperty1: A
+        },
+        filterWithB {
+          component: customFilter
+          customProperty1: B
+        }
+      }
+    }
+  }
+ */
+case class ComponentProviderConfig(provider: Option[String],
+                                   disabled: Boolean = false,
+                                    //TODO: do we need it?
+                                    elementPrefix: Option[String],
+                                    componentsConfig: Map[String, ComponentConfig],
```

Comment:
Why it need to be specified on this level? Why provider just not have raw `Config` and on their own decide how to internally handle it? The same notice regards to `elementPrefix` and `categories`. It is because you introduce standarized way to configure that? What about situation when someone will create provider that will have some groups of components and want to have configs for each groups of components?

Reply:
Yes, the main goal is to introduce some standardized way of handling it. 
We can discuss how far should it go, but I'd like to have some structure. Of course, each componentProvider will have its own config, but I think this should be just a list of fixed properties - I hope in the future ComponentProvider will also have list of "ConfigProperties" which admin will be able to configure via UI. 
It would be easier to design such UI if we had standard way of grouping etc.

Reply:
Ok, I know what was my main anxiety reason. I would like to:
- Have one common way to adding postprocessing of definition - we need to think about how this configuration to merge with current configuration of nodes, maybe we should move this configuration to existing configuration or other way?
- Make this configuration separate from core nk application, let's say that we have:
```
ComponentDefinitionPosprocessor {
  def postprocess(componentName): String
  def postprocess(param: Parameter): Parameter
  def postprocess(...) 
}
```
and it will be executed after each step of `GenericNodeContextTransformation`
If you have such an interface it is not important where you have such configuration and how you split it - if it will be configured for each component or it will be defined other way.

Reply:
I simplified initial design, component configuration refactor is deferred to next PRs...